Blockzilla/OverlayView.swift

```diff
@@ -401,12 +400,18 @@ class OverlayView: UIView {
         searchButtonGroup.first?.layer.cornerRadius = UIConstants.layout.suggestionViewCornerRadius
         searchButtonGroup.first?.clipsToBounds = true
         
-        if searchSuggestions.count == 1 {
-            searchButtonGroup.first?.layer.maskedCorners = findInPageButton.isHidden ? [.layerMaxXMinYCorner, .layerMinXMinYCorner, .layerMaxXMaxYCorner, .layerMinXMaxYCorner] : [.layerMaxXMinYCorner, .layerMinXMinYCorner]
-        } else {
-            searchButtonGroup.first?.layer.maskedCorners = [.layerMaxXMinYCorner, .layerMinXMinYCorner]
-            searchButtonGroup.last?.layer.maskedCorners = findInPageButton.isHidden ? [.layerMaxXMaxYCorner, .layerMinXMaxYCorner] : []
+        let searchSuggestionsPromptHidden = UserDefaults.standard.bool(forKey: SearchSuggestionsPromptView.respondedToSearchSuggestionsPrompt) || searchQuery.isEmpty
+        
+        var firstSearchButtonMaskedCorners: CACornerMask = [.layerMaxXMinYCorner, .layerMinXMinYCorner]
+        if !searchSuggestionsPromptHidden {
+            firstSearchButtonMaskedCorners = findInPageButton.isHidden ? [.layerMaxXMaxYCorner, .layerMinXMaxYCorner] : []
+        } else if searchSuggestions.count == 1 {
+            firstSearchButtonMaskedCorners = findInPageButton.isHidden ? [.layerMaxXMinYCorner, .layerMinXMinYCorner, .layerMaxXMaxYCorner, .layerMinXMaxYCorner] : [.layerMaxXMinYCorner, .layerMinXMinYCorner]
```

Comment:
Suggestion: It's pretty hard to read and notice which corners are masked. Maybe we can put the arrays in some variables that can be names `allCorners` `bottomCorners` etc.

Reply:
Ok, I will add this change. Thank you!